### PR TITLE
Make embassy::time optional

### PR DESCRIPTION
**Pull request by Dirbaio**

### CODE REVIEW

File: embassy-stm32/Cargo.toml [hunk 1]
```diff
@@ -6,7 +6,7 @@ edition = "2018"
 resolver = "2"
 
 [dependencies]
-embassy = { version = "0.1.0", path = "../embassy" }
+embassy = { version = "0.1.0", path = "../embassy", features = ["time-tick-32768hz"] }
```
Comment (thalesfragoso) [inline embassy-stm32/Cargo.toml:9]: Do we want to specify the ticks here ? Binary crates won't be able to choose the right tick for their job. However, `embassy-stm32` needs a tick, which means that we will need to provide tick feature in `embassy-stm32` too :/

**Review by thalesfragoso — COMMENTED** [review]
Thanks, I think this is a good idea overall, but not sure what is the best way to also give flexibility for the end user. I left a comment.

File: embassy-stm32/Cargo.toml (hunk 1 above)
Comment (Dirbaio) [reply embassy-stm32/Cargo.toml:9]: Yeah... When we do this, embassy-stm32 will need `time-tick-X` features which enable the corresponding ones in `embassy`...

Comment (Dirbaio) [reply embassy-stm32/Cargo.toml:9]: The other way around would be somehow embassy configuring the clock with the enabled tick rate, but some hardware can only work with one tick rate. For example nrf's can only do 32768hz. So I think it's better if the tick rate is enabled by the hal

Comment (thalesfragoso) [reply embassy-stm32/Cargo.toml:9]: And a default one...
Do you want to do this on this PR or in a later one ? If later, maybe we want to bump it to 1MHz ?

Comment (Dirbaio) [reply embassy-stm32/Cargo.toml:9]: The current stm32 clock can only do 32768hz right? We can add flexibility later. No opinion on the default from me, except as discussed on matrix 1mhz+16bit overflows too fast, we'd probably want 1mhz+32bit.

Comment (thalesfragoso) [reply embassy-stm32/Cargo.toml:9]: Thinking again, 32k seems a saner default.

**Review by thalesfragoso — APPROVED** [review]
Thanks!

**Pull request merged**